src/wallet/wallet.cpp

```diff
@@ -1837,9 +1837,13 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
         chain().findBlock(block_hash, FoundBlock().inActiveChain(block_still_active).nextBlock(FoundBlock().inActiveChain(next_block).hash(next_block_hash)));
 
         if (fetch_block) {
-            // Read block data
+            // Read block data and locator if needed (the locator is usually null unless we need to save progress)
             CBlock block;
-            chain().findBlock(block_hash, FoundBlock().data(block));
+            CBlockLocator loc;
```

Comment:
could we move this closer to its first usage instead of announcing it early?

Reply:
Sorry, which bit specifically? I can do if I have to retouch.

Reply:
the locator isn't needed for the next line, we could narrow its scope by moving it closer to its first usage